packages/docusaurus-theme-classic/src/theme/DocCardList/index.tsx

```diff
@@ -21,15 +23,18 @@ function filterItems(items: PropSidebarItem[]): PropSidebarItem[] {
   });
 }
 
-export default function DocCardList({
-  items,
-}: {
-  items: PropSidebarItem[];
-}): JSX.Element {
+export default function DocCardList({items, isGenerated}: Props): JSX.Element {
+  const columnSize = 6;
+  const columnCount = 12 / columnSize;
   return (
     <div className="row">
       {filterItems(items).map((item, index) => (
-        <article key={index} className="col col--6 margin-bottom--lg">
+        <article
+          key={index}
+          className={clsx(`col col--${columnSize}`, {
+            'margin-bottom--lg': !isGenerated,
+            'margin-top--lg': isGenerated && index + 1 > columnCount,
```

Comment:
I don't think this is the correct algorithm. It should be something closer to the lines of `:last-child`, otherwise if you have `<DocCardList>` as the last item of a Markdown page, you still get a larger margin:

<img width="839" alt="image" src="https://user-images.githubusercontent.com/55398995/162349602-33f7f711-808d-4ca0-9c27-d3d6fefa90e7.png">